src/view/pane/message.js

```diff
@@ -189,41 +190,44 @@ Candy.View.Pane = (function(self, $) {
         }
       });
 
-      var notifyEvtData = {
-        name: name,
-        displayName: Candy.Util.crop(name, Candy.View.getOptions().crop.message.nickname),
-        roomJid: roomJid,
-        message: message,
-        time: Candy.Util.localizedTime(timestamp),
-        timestamp: timestamp.toISOString()
-      };
-      /** Event: candy:view.message.notify
-       * Notify the user (optionally) that a new message has been received
-       *
-       * Parameters:
-       *   (Object) templateData - Template data consists of:
-       *                           - (String) name - Name of the sending user
-       *                           - (String) displayName - Cropped name of the sending user
-       *                           - (String) roomJid - JID into which the message was sent
-       *                           - (String) message - Message text
-       *                           - (String) time - Localized time of message
-       *                           - (String) timestamp - ISO formatted timestamp of message
-       */
-      $(Candy).triggerHandler('candy:view.message.notify', notifyEvtData);
-
-      // Check to see if in-core notifications are disabled
-      if(!Candy.Core.getOptions().disableCoreNotifications) {
-        // Notify the user about a new private message
-        if(Candy.View.getCurrent().roomJid !== roomJid || !self.Window.hasFocus()) {
-          self.Chat.increaseUnreadMessages(roomJid);
-          if(Candy.View.Pane.Chat.rooms[roomJid].type === 'chat' && !self.Window.hasFocus()) {
-            self.Chat.Toolbar.playSound();
+      if (!carbon) {
```

Comment:
At least one resource will receive a non-carbon version of the message. If the contact has resource locked, it will be that resource. If it has not, then all resources will receive a non-carbon version. This determination is made by the server, and allows carbon = silent to make sense.